### PR TITLE
Fix sporadic flowauth fails

### LINKED ISSUE #844: Sporadic flowauth test fails
FlowAuth tests are periodically failing on the first test, e.g. https://circleci.com/gh/Flowminder/FlowKit/59539 or see https://dashboard.cypress.io/#/projects/67obxt/runs/2933/failures/9f4670b4-f665-4edb-afa0-5f4410600139

This seems to be because cypress is _sometimes_ not sending the cookie.

### PR DESCRIPTION
**Pull request by greenape**
Closes #844

### I have:

- [x] Formatted any Python files with [black](https://github.com/ambv/black)
- [x] Brought the branch up to date with master
- [x] Added any relevant Github labels
- [x] Added tests for any new additions
- [ ] Added or updated any relevant documentation
- [ ] Added an Architectural Decision Record (ADR), if appropriate
- [ ] Added an [MPLv2 License Header](https://www.mozilla.org/en-US/MPL/headers/) if appropriate
- [x] Updated the [Changelog](https://github.com/Flowminder/FlowKit/blob/master/CHANGELOG.md) 

### Description

Added two multiprocessing `Event`s, which prevent more than one forked flowauth process from trying to initialise the database, or create demo data, and will cause others to wait until this is done.

### CODE REVIEW

File: flowauth/backend/tests/test_commands.py [hunk 1]
```diff
@@ -64,16 +64,73 @@ def test_init_db_force(app):
...
 
 
+def test_demo_data_only_sets_up_once(app, caplog):
+    """
+    Demo data should only be created once.
+    """
+    with app.app_context():
+        runner = app.test_cli_runner()
+        app.config["DB_IS_SET_UP"].clear()
+        result = runner.invoke(demodata)
```
Comment (maxalbert) [inline flowauth/backend/tests/test_commands.py]: Just to double-check: these two instances of `runner.invoke(demodata)` will run in serial, i.e. the first one will definitely finish before the second one is run? Just want to make sure this test can't accidentally fail if they run concurrently.

File: flowauth/backend/tests/test_commands.py [hunk 2]
```diff
@@ -64,16 +64,73 @@ def test_init_db_force(app):
...
+
+
+def test_demo_data_skips_when_in_progress(app, caplog):
+    """
+    Shouldn't attempt to create demo data if something else is.
+    """
+    with app.app_context():
+        runner = app.test_cli_runner()
+        app.config["DB_IS_SET_UP"].clear()
+        app.config["DB_IS_SETTING_UP"].set()
```
Comment (maxalbert) [inline flowauth/backend/tests/test_commands.py]: Technically the use of `DB_IS_SETTING_UP` as a flag is an internal implementation detail, but I can't think of an easy way to reliably trigger the test condition in a different way so I guess it's fine to use it in the test too.

Comment (greenape) [reply flowauth/backend/tests/test_commands.py]: Actually that's a good point - these should probably be on `g` instead really. I'll change it.

File: flowauth/backend/tests/test_commands.py (hunk 1 above)
Comment (greenape) [reply flowauth/backend/tests/test_commands.py]: Definitely serial.

**Pull request merged**